### PR TITLE
Use symlink xattr functions from x/sys/unix

### PR DESCRIPTION
**Pull request by tklauser**
Update x/sys/unix to get the symlink xattr syscall wrappers Lgetxattr and Lsetxattr (introduced in
golang/sys@b90f89a) and switch moby to use these instead of providing own syscall wrappers.

### CODE REVIEW

**Comment by tophj-ibm** [comment]
failure looks related. I think the changes in golang.org/x/sys/windows/ztypes_windows.go are breaking it.
```
07:47:01 # github.com/docker/docker/pkg/term
07:47:01 pkg/term/term_windows.go:82: constant 4294967286 overflows int
07:47:01 pkg/term/term_windows.go:88: constant 4294967285 overflows int
07:47:01 pkg/term/term_windows.go:94: constant 4294967284 overflows int
07:47:12 Build step 'Execute shell' marked build as failure
```

**Comment by tophj-ibm** [comment]
found the cl that changed it, might be useful https://go-review.googlesource.com/#/c/39608/

File: pkg/system/xattrs_linux.go
Comment (tklauser) [inline pkg/system/xattrs_linux.go]: Ok, will fold it in.

**Comment by tklauser** [comment]
@tophj-ibm Thanks for the hint. I currently see two possible solutions:

1. Switch GetStdFile in github.com/Azure/go-ansiterm/winterm to take an uint32 instead of int and check against windows.STD_*_HANDLE from golang.org/x/sys/windows.
2. (Temporarily) replace windows.STD_*_HANDLE in pkg/term/term_windows with either the values from syscall or define them locally as ints.

What would be the preferred way to resolve this? Would it be acceptable to implement option 2 as part of this PR and implement 1, once a fix is merged in github.com/Azure/go-ansiterm/winterm. Or does anyone see another option?

File: pkg/system/xattrs_linux.go
Comment (tklauser) [inline pkg/system/xattrs_linux.go]: @AkihiroSuda I guess you meant to remove the Lsetxattr wrapper function? The Lgetxattr wrapper has additional functionality (resizing the buffer if the xattrs don't fit) which is not covered by the version from x/sys/unix.

**Comment by tophj-ibm** [comment]
@tklauser I am not a maintainer (but @AkihiroSuda is), but IMO option 2 where you switch the file to use syscall instead of golang.org/x/sys/windows and then open an issue for the first problem that seems good to me. 

I don't know if they will do anything about it in the go-ansiterm repo because they would have to switch their entire project from syscall to /x/sys/ in order to fix it, so the temporary solution might be sticking around for a while.

File: pkg/system/xattrs_linux.go
Comment (AkihiroSuda) [inline pkg/system/xattrs_linux.go]: ok, makes sense

Comment (tklauser) [inline pkg/system/xattrs_linux.go]: Just checked. The wrapper is still necessary, as there is also a version in pkg/system/xattrs_unsupported.go for !linux.

**Comment by tklauser** [comment]
@AkihiroSuda Just pushed another version with an additional commit implementing option 2 outlined in https://github.com/moby/moby/pull/33539#issuecomment-306709227 (temporarily switching to use STD_*_HANDLE from syscall) as @tophj-ibm suggested. All builds looks fine now.

**Comment by tophj-ibm** [comment]
changes SGTM

File: pkg/term/term_windows.go
Comment (thaJeztah) [inline pkg/term/term_windows.go]: Should we mark this with a `TODO` ?

**Comment by thaJeztah** [comment]
@tklauser looks like this needs a rebase 😢

File: pkg/term/term_windows.go
Comment (tklauser) [inline pkg/term/term_windows.go]: Makes sense, will add a `TODO`.

**Comment by tklauser** [comment]
@thaJeztah, rebased upon latest master and added the TODO as mentioned in https://github.com/moby/moby/pull/33539#discussion_r121666531

**Comment by tklauser** [comment]
Rebased upon latest master again to resolve merge conflicts

**Comment by thaJeztah** [comment]
ping @johnstep PTAL

sadly needs another rebase @tklauser  - I'll try to get someone review this ASAP, otherwise ping me 👍

**Comment by tklauser** [comment]
Rebased again, but sadly the Windows build now fails due to the change in x/sys/windows introduced by  golang/sys@ff24cb3cd86e67842e0d078a5e0cb92d9ffcf8b4 (https://go-review.googlesource.com/c/39608/)

Any advise on how to properly fix this?

**Comment by tklauser** [comment]
Just pushed an additional commit adding an explicit type conversion in the call to `windows.GetStdHandle`. Hope that is OK...

**Comment by tophj-ibm** [comment]
changes still seem fine to me.

**Review by tophj-ibm — APPROVED** [review]

**Comment by thaJeztah** [comment]
For a list of commits in the golang.org/x/sys bump; https://github.com/golang/sys/compare/8f0908ab3b2457e2e15403d3697c9ef5cb4b57a9...739734461d1c916b6c72a63d7efda2b27edb369f

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by dnephin — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
all green - merging. Thanks @tklauser !

**Pull request merged**
